Source/WebCore/dom/Document.cpp

```diff
@@ -9014,31 +9014,41 @@ HTMLDialogElement* Document::activeModalDialog() const
 
 HTMLElement* Document::topmostAutoPopover() const
 {
-    for (auto& element : makeReversedRange(m_topLayerElements)) {
-        if (auto* candidate = dynamicDowncast<HTMLElement>(element.get()); candidate && candidate->popoverState() == PopoverState::Auto)
-            return candidate;
-    }
-
-    return nullptr;
+    if (m_popoverStack.isEmpty())
+        return nullptr;
+    return m_popoverStack.last().get();
 }
 
 // https://html.spec.whatwg.org/#hide-all-popovers-until
 void Document::hideAllPopoversUntil(Element* endpoint, FocusPreviousElement focusPreviousElement, FireEvents fireEvents)
 {
-    Vector<Ref<HTMLElement>> popoversToHide;
-    for (auto& item : makeReversedRange(m_topLayerElements)) {
-        if (!is<HTMLElement>(item))
-            continue;
-        auto& element = downcast<HTMLElement>(item.get());
-        if (element.popoverState() == PopoverState::Auto) {
-            if (&element == endpoint)
+    auto closeAllOpenPopovers = [&]() {
+        while (auto* popover = topmostAutoPopover())
```

Comment:
Would be nice to use `RefPtr` for `popover`.

Reply:
Done.